spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarAdministration.java

```diff
@@ -0,0 +1,171 @@
+	}
+
+	private PulsarAdmin createAdminClient() throws PulsarClientException {
+		return adminBuilder.build();
+	}
+
+	@Override
+	public void createOrModifyTopics(PulsarTopic... topics) {
+		PulsarAdmin admin;
+		try {
+			admin = createAdminClient();
+		} catch (Exception e) {
+			throw new IllegalStateException("Could not create PulsarAdmin", e);
+		}
+
+		if (admin != null) {
+			createOrModifyTopicsIfNeeded(admin, Arrays.asList(topics));
+		}
+	}
+
+	private Map<String, List<PulsarTopic>> getTopicsPerNamespace(Collection<PulsarTopic> topics) {
+		Map<String, List<PulsarTopic>> topicsPerNamespace = new HashMap<>();
+		topics.forEach(topic -> {
+			PulsarTopic.TopicComponents topicComponents = topic.getComponents();
+			String tenant = topicComponents.tenant();
+			String namespace = topicComponents.namespace();
+			String namespaceIdentifier = tenant + "/" + namespace;
+			topicsPerNamespace.computeIfAbsent(namespaceIdentifier, k -> new ArrayList<>()).add(topic);
+		});
+		return topicsPerNamespace;
+	}
+
+	private List<String> getMatchingTopicPartitions(PulsarTopic topic, List<String> existingTopics) {
+		return existingTopics
+				.stream()
+				.filter(existing -> existing.startsWith(topic.getFullyQualifiedTopicName() + "-partition-"))
+				.toList();
+	}
+
+	private void createOrModifyTopicsIfNeeded(PulsarAdmin admin, Collection<PulsarTopic> topics) {
+		if (CollectionUtils.isEmpty(topics)) {
+			return;
+		}
+
+		Map<String, List<PulsarTopic>> topicsPerNamespace = getTopicsPerNamespace(topics);
+
+		Set<PulsarTopic> topicsToCreate = new HashSet<>();
+		Set<PulsarTopic> topicsToModify = new HashSet<>();
+
+		topicsPerNamespace.forEach((namespace, requestedTopics) -> {
+			try (admin) {
+				List<String> existingTopicsInNamespace = admin.topics().getList(namespace);
+
+				for (PulsarTopic topic : requestedTopics) {
+					if (topic.isPartitioned()) {
+						List<String> matchingPartitions = getMatchingTopicPartitions(topic, existingTopicsInNamespace);
+						if (matchingPartitions.isEmpty()) {
+							logger.info("Topic " + topic.getFullyQualifiedTopicName() + " does not exist.");
+							topicsToCreate.add(topic);
+						} else {
+							int numberOfExistingPartitions = matchingPartitions.size();
+							if (numberOfExistingPartitions < topic.numberOfPartitions()) {
+								logger.info("Topic " + topic.getFullyQualifiedTopicName() + " found with "
+										+ numberOfExistingPartitions + " partitions.");
+								topicsToModify.add(topic);
+							} else if (numberOfExistingPartitions > topic.numberOfPartitions()) {
+								throw new IllegalStateException(
+										"Topic " + topic.getFullyQualifiedTopicName() + " found with "
+												+ numberOfExistingPartitions
+												+ " partitions. Needs to be deleted first.");
+							}
+						}
+					} else {
+						if (!existingTopicsInNamespace.contains(topic.getFullyQualifiedTopicName())) {
+							logger.info("Topic " + topic.getFullyQualifiedTopicName() + " does not exist.");
```

Comment:
Ahh yeh, not enough detail in my initial comment. 

So, prior to log accessor, if I wanted to conditionally debug log to avoid possible expensive log message creation I would have to pollute the code w/ `if logger.isXYZEnabled()` which made the code terse and I would often not use it in favor of more readable code. 
```
if (logger.isDebugEnabled()) {
	logger.debug("expensive string goes here");
}
```

However, the beauty of LogAccessor is this API to pass in supplier and under the hood it does the `if logger.isXYZEnabled()` and solves both clean code and better performing tradeoff of the previous logger APIs. 
```java
logger.debug(() -> "expensive string goes here");
```

Reply:
By "lambda" I really meant "the log accessor API that takes a supplier."